packages/netlify-cms-widget-file/src/withFileControl.js

```diff
@@ -118,9 +108,9 @@ export default function withFileControl({ forImage } = {}) {
 
     shouldComponentUpdate(nextProps) {
       /**
-       * Always update if the value changes.
+       * Always update if the value or getAsset changes.
        */
-      if (this.props.value !== nextProps.value) {
+      if (this.props.value !== nextProps.value || this.props.getAsset !== nextProps.getAsset) {
```

Comment:
Some performance issues here, `getAsset` is a function and between props updates, the above check will always pass. Editing any other widget will always force this widget to re-render.

Reply:
Actually though about that and we could memoize it here https://github.com/netlify/netlify-cms/blob/8d67de0e681e02ddbf811eecc5e4744b324b7e61/packages/netlify-cms-core/src/actions/media.ts#L85 if we it causes issue